### PR TITLE
Pass (selected) Lighthouse settings via netlify.toml

### LINKED ISSUE #387: Allow configuring the device to monitor (Desktop / Mobile)
I did not see in the docs a way to configure the device, only by category (performance, SEO, etc). 

image

I think that by default is mobile, but right now, I just want to monitor only in Desktop. Is this possible?

image

### PR DESCRIPTION
**Pull request by jackbrewer**
Closes #387

This PR adds the ability to start passing custom Lighthouse settings, exposing some as optional `netlify.toml` fields.

Example usage (also added to the README):
```
[[plugins]]
  package = "@netlify/plugin-lighthouse"

  [plugins.inputs.settings]
    preset = "desktop" # Optionally run Lighthouse using a desktop configuration
    locale = "es" # Any Lighthouse-supported locale, used to generate reports in a different language
```

To start, the `preset` setting allows developers to opt-in to a desktop configuration (which are already set up by Lighthouse).

As a quick-win extra, I've also exposed the `locale` setting. This will allow devs to optionally specify the language used in the generated report. It also affects the reported scores within the Netlify UI. As with all `.toml` settings, this will apply at a site level, not per-user. Changing language of the reports will affect the output for all viewers of the deploy detail report.

Partial screenshot of a report generated using the above options:
image

### CODE REVIEW

File: src/settings.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { getSettings } = require('./settings');
+
+describe('replacements', () => {
+  it('should return nothing with no settings set', () => {
+    expect(getSettings()).toEqual(undefined);
+    expect(getSettings({})).toEqual(undefined);
```
Comment (aitchiss) [inline src/settings.test.js:6]:
```suggestion
    expect(getSettings()).toBeUndefined();
    expect(getSettings({})).toBeUndefined();
```
[dust] just a thought 🤷‍♀️

**Comment by jackbrewer** [comment]
A note on the `preset` naming, taken from [this document](https://github.com/GoogleChrome/lighthouse/blob/main/docs/emulation.md#emulation-in-lighthouse): 

> With the default configuration, Lighthouse emulates a mobile device. There's [a desktop configuration](https://github.com/GoogleChrome/lighthouse/blob/main/core/config/desktop-config.js), available to CLI users with --preset=desktop, which applies a consistent desktop environment and scoring calibration.

I originally planned for `device = "mobile"` or `device = "desktop"`, but as mobile is the default, it's a redundant option. When implementing `device = "desktop"`, I noticed I was essentially doing the exact same as the existing `preset=desktop` Lighthouse option, so switched to that. `preset` also has a couple of other options (`perf` or `experimental`) which the UI isn't ready to handle, but it could be an option in future.

File: src/settings.test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const { getSettings } = require('./settings');
+
+describe('replacements', () => {
+  it('should return nothing with no settings set', () => {
+    expect(getSettings()).toEqual(undefined);
```
[the remark this answers is not in the document]
Comment (jackbrewer) [reply src/settings.test.js:5]: One is no input `()`, the other is and empty object `({})` – I'll separate them to be clearer!

File: src/settings.test.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    expect(getSettings({})).toEqual(undefined);
+  });
+
+  it('should return a template config with preset set to desktop', () => {
+    const derivedSettings = getSettings({ preset: 'desktop' });
+    expect(derivedSettings.extends).toEqual('lighthouse:default');
+    expect(derivedSettings.settings.formFactor).toEqual('desktop');
```
[the remark this answers is not in the document]
Comment (jackbrewer) [reply src/settings.test.js:12]: Yep, it gets set when we choose the [desktop config](https://github.com/GoogleChrome/lighthouse/blob/main/core/config/lr-desktop-config.js#L15)

**Review by aitchiss — APPROVED** [review]
Works as described! Love to see more customisation being possible ✨

**Review by kevinsamoei — APPROVED** [review]

**Pull request merged**